### PR TITLE
ttbb datacards for 2017 mc production

### PR DESCRIPTION
**Pull request by Andrej-CMS**
please check the card for 4FNS ttbb Sherpa production

### CODE REVIEW

**Comment by Andrej-CMS** [comment]
Now also added the cards for MG5aMC@NLO ttbb_4f sample

File: bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_madspin_card.dat [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#*                                                          *
+#************************************************************
+#Some options (uncomment to apply)
+#
+#directory for gridpack mode
+set ms_dir ./madspingrid
+
+#initialization parameters
+ set Nevents_for_max_weigth 250 # number of events for the estimate of the max. weight
+ #set BW_cut 15                # cut on how far the particle can be off-shell
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_madspin_card.dat]: please cleanup commented commands

File: bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#import model loop_sm
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_proc_card.dat]: please cleanup commented inputs

File: bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#***********************************************************************
+    1   = lpp1    ! beam 1 type (0 = no PDF)
+    1   = lpp2    ! beam 2 type (0 = no PDF)
+ 6500   = ebeam1  ! beam 1 energy in GeV
+ 6500   = ebeam2  ! beam 2 energy in GeV
+#***********************************************************************
+# PDF choice: this automatically fixes also alpha_s(MZ) and its evol.  *
+#***********************************************************************
+ lhapdf    = pdlabel   ! PDF set                                     
+ 320900    = lhaid     ! if pdlabel=lhapdf, this is the lhapdf number
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_run_card.dat]: please use
$DEFAULT_PDF_SETS = lhaid

File: bin/Sherpa/cards/production/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV [hunk 4]
```diff
@@ -26,15 +26,14 @@
...
   LOOPGEN:=OpenLoops;
 
   % settings relevant for 4-flavour scheme
   MASSIVE[5]=1
-  MASS[5]=4.75
+  MASS[5]=4.8
   PDF_LIBRARY LHAPDFSherpa
   #LHAPDF_GRID_PATH=./PDFsets
-  PDF_SET NNPDF30_nnlo_as_0118_nf_4
+  PDF_SET NNPDF31_nnlo_as_0118_nf_4
```
Comment (perrozzi) [inline bin/Sherpa/cards/production/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV]: please copy this datacard (don't remove the old one) to 
bin/Sherpa/cards/production/2017/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV

File: bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_run_card.dat (hunk 3 above)
Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31/ttbb_4f_ckm_NLO_NNPDF31_run_card.dat]: cfr https://twiki.cern.ch/twiki/bin/view/CMS/QuickGuideMadGraph5aMCatNLO#PDF_Choice_for_2017_production

File: bin/Sherpa/cards/production/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV_NNPDF31 [hunk 5]
```diff
@@ -0,0 +1,59 @@
+(run){
```
Comment (perrozzi) [inline bin/Sherpa/cards/production/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV_NNPDF31]: can you please create a folder called 2017
in/Sherpa/cards/production/**2017**/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV_NNPDF31

**Comment by perrozzi** [comment]
Thanks. One last thing...
As for this PR https://github.com/cms-sw/genproductions/pull/1360 a question arises on a convention for a folder structure to collect similar processes to avoid the proliferation of thousands of folders in one main folder.
For POWHEG this is trivial as there are predefined processes, but for mg5_amc is less trivial.
In your case I would think of a TTplusX main folder, like:
[...]/cards/production/2017/TTplusX/ttbb_4f_ckm_NLO_NNPDF31/
@efeyazgan what do you think?

**Comment by perrozzi** [comment]
(for Sherpa the amount of datacards is not at the level of mg5_amc so it's less critical for the moment...)

**Comment by efeyazgan** [comment]
I think TTplusX is a good idea for this case.

**Comment by perrozzi** [comment]
@Andrej-CMS there must have been a misunderstanding because is more important for mg5_amc than for Sherpa, while you did it only for Sherpa and not for mg5_amc...

**Comment by Andrej-CMS** [comment]
@perrozzi 
Just to make sure, I push the right thing this time. Should I also move the other tt+jets sample into the TTPlusX folder

**Comment by perrozzi** [comment]
genproductions/bin/MadGraph5_aMCatNLO/cards/production/2017/ttbb_4f_ckm_NLO_NNPDF31 ---> genproductions/bin/MadGraph5_aMCatNLO/cards/production/2017/**TTplusX**/ttbb_4f_ckm_NLO_NNPDF31

**Comment by perrozzi** [comment]
looks ok to me now, apart from
`bin/Sherpa/cards/production/13TeV/higgs/Run.dat_ttbb_4FS_OpenLoops_13TeV 100755 → 100644`
I hope it doesn't create any permission problem, @pmillet can you comment?
@efeyazgan to you to merge when you want

**Comment by pmillet** [comment]
@perrozzi sorry, comment on what?

**Comment by perrozzi** [comment]
in this PR there is a spuroius change of permissions 100755 → 100644 for a sherpa card, I wonder if is still ok or is better to keep the original ones

**Comment by pmillet** [comment]
I think since its only a datacard / textfile the execute bit should not matter.

**Comment by perrozzi** [comment]
ok thanks, ready to merge from my side then

**Comment by efeyazgan** [comment]
@Andrej-CMS , which is the central set for the sherpa one?

**Comment by Andrej-CMS** [comment]
@efeyazgan 
NNPDF31_nnlo_as_0118_nf_4 or should we go back to NNPDF30 NLO?

**Comment by efeyazgan** [comment]
As far as I understand NNPDF30 is used for the default sherpa tune. So, we will need to use it until we have a cms dedicated tune with nnpdf3.1 for sherpa parton shower. @pmillet , please comment.

**Comment by pmillet** [comment]
Here are two statements from Marek I got concerning this on the past:

- "I would also like ask you to use the default PDF, NNPDF30, as this is the one the non-perturbative parameters of Sherpa-2.2.x are tuned to."

- " All Sherpa-2.2.x releases share the same default PDF and accompanying soft-physics tune. The PDF is NNPDF3.0 NNLO with as(mZ)=0.118. We have an internal direct interface to this PDF, but its LHAPDF equivalent is NNPDF30_nnlo_as_0118 or the old-style ID 261000."

Thus I think we should stay with Sherpas default PDF  for now (NNPDF30).

**Pull request merged**